hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java

```diff
@@ -0,0 +1,265 @@
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.virtualmap.VirtualKey;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+/** Represents a key for a unique token (NFT). */
+public class UniqueTokenKey implements VirtualKey<UniqueTokenKey> {
+    private static final long CLASS_ID = 0x17f77b311f6L;
+
+    /** Current version of the encoding scheme. */
+    /* package */ static final int CURRENT_VERSION = 1;
+
+    /**
+     * Expected maximum number of bytes this class will serialize to. Serialization format will be:
+     * - (1 byte) number of bytes for next two fields: - The higher 4-bits will contain a number (0
+     * - 8) indicating number of bytes needed for the entity number. - The lower 4-bits will contain
+     * a number (0 - 8) indicating number of bytes needed for the serial number. - (variable, 0 to 8
+     * bytes) the non-leading zero-bytes representing the entity number. - (variable, 0 to 8 bytes)
+     * the non-leading zero-bytes representing the serial number.
+     */
+    public static final int ESTIMATED_SIZE_BYTES = Long.BYTES + Long.BYTES + 1;
+
+    /**
+     * Constructs a UniqueTokenKey from an NftId instance.
+     *
+     * @param nftId the NftId to create a UniqueTokenKey from.
+     * @return a new instance of a UniqueTokenKey corresponding to NftId.
+     */
+    public static UniqueTokenKey from(final NftId nftId) {
+        return new UniqueTokenKey(nftId.num(), nftId.serialNo());
+    }
+
+    public static UniqueTokenKey from(final NftNumPair nftNumPair) {
+        return new UniqueTokenKey(nftNumPair.tokenNum(), nftNumPair.serialNum());
+    }
+
+    /** The entity number of the token. */
+    private long entityNum;
+
+    /** Serial number of the token. */
+    private long tokenSerial;
+
+    /** Hashcode will be updated whenever tokenNum changes. */
+    private int hashCode;
+
+    public UniqueTokenKey() {}
+
+    public UniqueTokenKey(long entityNum, long tokenSerial) {
+        setTokenId(entityNum, tokenSerial);
+    }
+
+    public long getNum() {
+        return entityNum;
+    }
+
+    public long getTokenSerial() {
+        return tokenSerial;
+    }
+
+    private void setTokenId(long entityNum, long tokenSerial) {
+        this.entityNum = entityNum;
+        this.tokenSerial = tokenSerial;
+        // Consider using NonCryptographicHashing.hash64(long1, long2) when made available.
+        this.hashCode = Objects.hash(entityNum, tokenSerial);
+    }
+
+    private static int computeNonZeroBytes(long value) {
+        // The value returned from this will range in [0, 8].
+        if (value == 0) {
+            return 0;
+        }
+
+        // Max value here is (64 - 0)/8 = 8
+        // Min value here is ceil((64 - 63)/8) = 1
+        return (int) Math.ceil((Long.SIZE - Long.numberOfLeadingZeros(value)) / 8D);
```

Comment:
Leemon prefers we avoid floating-point arithmetic everywhere out of maximum paranoia. 😁 

Would an integral-only equivalent be,
```
final var nonZeroBits = Long.SIZE - Long.numberOfLeadingZeros(value);
return (nonZeroBits / 8) + Math.min(1, nonZeroBits % 8);
```